io/io/test/TROMemFileTests.cxx

```diff
@@ -0,0 +1,65 @@
+#include "TMemFile.h"
+
+#include "TError.h"
+#include "TMemFile.h"
+#include "TTree.h"
+#include <cstring>
+
+#include "gtest/gtest.h"
+
+static std::shared_ptr<const std::vector<char>> CreateBuffer(const char* title) {
+   TNamed n("name", title);
+   TMemFile memFile("a.root", "RECREATE", "TMemFile shared data test file", 0 /*no compression*/);
+   memFile.WriteTObject(&n);
+   memFile.Write();
+
+   std::vector<char> data;
+   data.resize(memFile.GetSize());
+   memFile.CopyTo(&data.front(), data.size());
+   return std::make_shared<const std::vector<char>>(std::move(data));
+}
+
+TEST(TROMemFile, Basics)
+{
+   constexpr const char title[] = "This is a title for TMemFile shared data test Basics";
+   std::shared_ptr<const std::vector<char>> dataPtr(CreateBuffer(title));
+
+   TMemFile rosmf("romemfile.root", dataPtr);
+   TObject *readN = rosmf.Get("name");
+   ASSERT_NE(nullptr, readN);
+   EXPECT_STREQ(title, readN->GetTitle());
+}
+
+TEST(TROMemFile, NoWriting)
+{
+   constexpr const char title[] = "This is a title for TMemFile shared data test NoWriting";
+   std::shared_ptr<const std::vector<char>> dataPtr(CreateBuffer(title));
+
+   TMemFile rosmf("romemfile.root", dataPtr);
+   auto oldIgnoreLevel = gErrorIgnoreLevel;
+   gErrorIgnoreLevel = kBreak;
+   TNamed doNotWrite("doNotWrite", "doNotWrite Title");
+   EXPECT_EQ(0, rosmf.WriteTObject(&doNotWrite));
+   gErrorIgnoreLevel = oldIgnoreLevel;
+}
+
+/// Check that TMemFile uses the original buffer, not a copy (like TMemFile)
```

Comment:
It seems that the comment may have been victim to search&replace idiosyncrasy.  Could you review it?